### PR TITLE
ghwatch: Adapt to terminal size changes

### LINKED ISSUE #78: ghwatch: Adapt to terminal size changes
Currently, the table size is computed once at start of the program. If the terminal resizes later (especially if it became smaller), it will keep generating it at those dimensions, which might result in display corruption. It would be nice to handle this.

### PR DESCRIPTION
**Pull request by Kvaciral**
Closes #78

### CODE REVIEW

**Comment by laanwj** [comment]
Thank you ! will test

**Comment by jonatack** [comment]
Nice! Will test soon.

**Comment by laanwj** [comment]
Tested and code review ACK 671d8147625eba52b026d4e7e0a7f94435c5140d

File: ghwatch.py [hunk 1]
```diff
@@ -301,31 +269,75 @@ def handle_mouse_click(b, config):
...
+                continue
+            if k == 'escape':
+                break
+            # print(k, Key.mouse_pos)
+            if k == 'mouse_click':
+                # TODO: highlight button when clicked for a bit of feedback?
+                for b in buttons:
+                    if b.x0 <= Key.mouse_pos[0] < b.x1 and b.y0 <= Key.mouse_pos[1] < b.y1:
+                        handle_mouse_click(b, config)
+
```
Comment (jonatack) [inline ghwatch.py:335]:
```suggestion

```

File: termlib/input.py [hunk 2]
```diff
@@ -165,6 +167,11 @@ def break_wait(cls):
         sleep(0.01)
         cls.new.clear()
 
+    @classmethod
+    def _resize_handler(cls, signum, frame):
+        cls.list.append('resize')
+        cls.new.set()
+
```
Comment (jonatack) [inline termlib/input.py:174]:
```suggestion

```

**Review by jonatack — COMMENTED** [review]
Concept ACK!

On my setup (debian, tilix with a few buffers open), this patch does adjust to a buffer resize, but with several rapid redraws (say, 3 to 7) before stabilizing. If it could do it with one redraw that would be great.

**Comment by jonatack** [comment]
(feel free to ignore if it's only me or not easy to do)

**Comment by laanwj** [comment]
> On my setup (debian, tilix with a few buffers open), this patch does adjust to a buffer resize, but with several rapid redraws (say, 3 to 7) before stabilizing. If it could do it with one redraw that would be great.

I think the underlying problem here is that redraws are currently so expensive because they involve github API queries. It would be better to separate this into a `load` function that queries github and stores the data, and `draw` would only draw the table from stored data (potentially doing sorting first).

I don't think it's necessary in this PR. Resizing should be fairly rare anyway :smile:

**Comment by jonatack** [comment]
> I think the underlying problem here is that redraws are currently so expensive because they involve github API queries. It would be better to separate this into a `load` function that queries github and stores the data, and `draw` would only draw the table from stored data (potentially doing sorting first).

Great point.

> I don't think it's necessary in this PR. 

Agree!

**Comment by jonatack** [comment]
Trailing whitespace removed in latest push per `git diff 671d814 d567991`

**Comment by Kvaciral** [comment]
Oops, missed that one: added line in termlib/input.py between two methods (the new _resize_handler() and _get_key() in the Key class)

**Pull request merged**